jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MojoCommon.java

```diff
@@ -94,15 +94,25 @@ static void finishUpdateChecker(
    * @param jibPluginConfiguration the build configuration
    * @return the list of resolved extra directories
    */
-  static List<Path> getExtraDirectories(JibPluginConfiguration jibPluginConfiguration) {
-    List<Path> paths = jibPluginConfiguration.getExtraDirectories();
-    if (!paths.isEmpty()) {
-      return paths;
+  static List<JibPluginConfiguration.ExtraDirectoryParameters> getExtraDirectories(
```

Comment:
Not necessary, but `import ... JibPluginConfiguration.ExtraDirectoryParameters` to import the nested type may work?